internal/cli/cmd/plugin/build/build.go

```diff
@@ -81,22 +98,185 @@ func (o *option) Execute() error {
 		return fmt.Errorf("unable to read manifest: %w", err)
 	}
 
-	// Then we need to create the archive.
-	// The archive contains the following file:
+	// Check if we need to vendor the dependencies:
+	// - Parse the module file into a CUE instance
+	ctx := cuecontext.New()
```

Comment:
I think this change should go to a dedicated function. It will isolate it from the rest of the code.

Regarding the defer function, I would say the code can return the function to be defer. And then, in the function `Execute` you can call the defer function returned in the `defer` state.

Reply:
Also, you will see with the changes from #2700 that you can now check if `schema` are required.

So all of that is under the condition that schemas are required. You should add this check at the beginning.

Reply:
Actually I wanted to encapsulate that in a dedicated function but was kinda stuck because of the defer, I didnt think about this trick thanks!